app/src/main/java/org/greenstand/android/TreeTracker/viewmodels/TreeHeightViewModel.kt

```diff
@@ -45,38 +44,27 @@ class TreeHeightViewModel(
             newTree
                 ?.let { tree ->
                     withContext(Dispatchers.IO) {
-
-                        val createTreeParams = CreateTreeParams(
-                            planterCheckInId = tree.planterCheckInId,
-                            photoPath = tree.photoPath,
-                            content = tree.content,
-                            treeUuid = tree.treeUuid
-                        )
-
-                        val treeId = createTreeUseCase.execute(createTreeParams)
-
-                        fun addKeyValueAttribute(key: String, value: String) {
-                            val entity =
-                                TreeAttributeEntity(
-                                    key = key,
-                                    value = value,
-                                    treeCaptureId = treeId
-                                )
-                            dao.insertTreeAttribute(entity)
-                        }
-
-                        with(TreeHeightAttributes(treeId = treeId, heightColor = treeColor!!)) {
-                            addKeyValueAttribute(TreeManager.TREE_COLOR_ATTR_KEY, heightColor.value)
-                            addKeyValueAttribute(TreeManager.APP_BUILD_ATTR_KEY, appBuild)
-                            addKeyValueAttribute(TreeManager.APP_FLAVOR_ATTR_KEY, appFlavor)
-                            addKeyValueAttribute(TreeManager.APP_VERSION_ATTR_KEY, appVersion)
+                        with(TreeHeightAttributes(heightColor = treeColor!!)) {
+                            tree.addTreeAttribute(
+                                Tree.Attributes.TREE_COLOR_ATTR_KEY, heightColor.value)
+                            tree.addTreeAttribute(
+                                Tree.Attributes.APP_BUILD_ATTR_KEY, appBuild)
+                            tree.addTreeAttribute(
+                                Tree.Attributes.APP_FLAVOR_ATTR_KEY, appFlavor)
+                            tree.addTreeAttribute(
+                                Tree.Attributes.APP_VERSION_ATTR_KEY, appVersion)
                         }
+                        createTreeUseCase.execute(tree)
                     }
                 }
                 ?.also {
                     toastMessageLiveData.postValue(R.string.tree_saved)
                     analytics.treeHeightMeasured(treeColor!!)
                     onFinishedLiveData.postValue(Unit)
+                    // Assign the current absolute step count to 'absoluteStepCountOnTreeCapture'
+                    // variable to calculate future step count deltas
+                    stepCounter.absoluteStepCountOnTreeCapture = stepCounter.absoluteStepCount
+                    stepCounter.disable()
```

Comment:
Same operations to collect data and unregister the step count event listener is made for the use case where Tree Height attribute feature is enabled in the App